tests/e2e/parallel/main_test.go

```diff
@@ -2138,72 +2138,211 @@ func TestScanSettingBinding(t *testing.T) {
+		t.Fatalf("expected rule %s to exist in namespace %s", unChangedTypeRuleName, origPb.Namespace)
+	}
+
+	tpMix := &compv1alpha1.TailoredProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tpMixName,
 			Namespace: f.OperatorNamespace,
 		},
 		Spec: compv1alpha1.TailoredProfileSpec{
-			Title:       "TestCisForE2EPool",
-			Description: "TestCisForE2EPool",
-			Extends:     "ocp4-cis",
-			SetValues: []compv1alpha1.VariableValueSpec{
+			Title:       "TestForManyRules",
+			Description: "TestForManyRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
 				{
-					Name:      "ocp4-var-role-master",
-					Rationale: "targets the e2e pool",
-					Value:     "e2e",
+					Name:      changeTypeRuleName,
+					Rationale: "this rule should be removed from the profile",
 				},
 				{
-					Name:      "ocp4-var-role-worker",
-					Rationale: "targets the e2e pool",
-					Value:     "e2e",
+					Name:      unChangedTypeRuleName,
+					Rationale: "this rule should not be removed from the profile",
 				},
 			},
 		},
 	}
 
-	createTPErr := f.Client.Create(context.TODO(), tp, nil)
-	if createTPErr != nil {
-		t.Fatal(createTPErr)
-	}
-	defer f.Client.Delete(context.TODO(), tp)
-
-	scanSettingBinding := compv1alpha1.ScanSettingBinding{
+	tpSingle := &compv1alpha1.TailoredProfile{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      scanSettingBindingName,
+			Name:      tpSingleName,
 			Namespace: f.OperatorNamespace,
 		},
-		Profiles: []compv1alpha1.NamedObjectReference{
-			{
-				Name:     tpName,
-				Kind:     "TailoredProfile",
-				APIGroup: "compliance.openshift.io/v1alpha1",
+		Spec: compv1alpha1.TailoredProfileSpec{
+			Title:       "TestForManyRules",
+			Description: "TestForManyRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      changeTypeRuleName,
+					Rationale: "this rule should not be removed from the profile",
+				},
 			},
 		},
-		SettingsRef: &compv1alpha1.NamedObjectReference{
-			Name:     "e2e-default",
-			Kind:     "ScanSetting",
-			APIGroup: "compliance.openshift.io/v1alpha1",
-		},
 	}
 
-	err := f.Client.Create(context.TODO(), &scanSettingBinding, nil)
+	createTPErr := f.Client.Create(context.TODO(), tpMix, nil)
+	if createTPErr != nil {
+		t.Fatal(createTPErr)
+	}
+	defer f.Client.Delete(context.TODO(), tpMix)
+	// check the status of the TP to make sure it has no errors
+	err = f.WaitForTailoredProfileStatus(f.OperatorNamespace, tpMixName, compv1alpha1.TailoredProfileStateReady)
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer f.Client.Delete(context.TODO(), &scanSettingBinding)
 
-	// it's enough to check that the scan finishes
-	err = f.WaitForSuiteScansStatus(f.OperatorNamespace, scanSettingBindingName, compv1alpha1.PhaseDone, compv1alpha1.ResultNonCompliant)
+	hasRule, err := f.EnableRuleExistInTailoredProfile(f.OperatorNamespace, tpMixName, changeTypeRuleName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !hasRule {
+		t.Fatal("Expected the tailored profile to have the rule")
```

Comment:
Follow up nit: We could add some logging here so it's more clear what rule is missing from what profile.